snappy_pipeline/workflows/somatic_variant_calling/scripts/split_region.py

```diff
@@ -0,0 +1,60 @@
+import csv
+from bisect import bisect
+from contextlib import redirect_stderr
+from itertools import accumulate
+
+from snappy_wrappers.tools.genome_windows import matches_any
+
+
+def split_regions(
+    contig_lengths: dict[str, int], n_chunks: int, offset: int = 1
+) -> list[list[[str, int, int]]]:
+    """Splits a given collection of contigs based on their length into `n_chunks` equal sized chunks"""
+    contigs, lengths = zip(*contig_lengths.items())
+    cumulative_lengths = list(accumulate(lengths, initial=0))
+    chunk_size = cumulative_lengths[-1] // n_chunks
```

Comment:
Is that correct? I have the feeling that the code above would return a chunk size of 4 for 2 chunks if the cumulative length is 9.
I would suggest:
```
chunk_size = 1 + (cumulative_lengths[-1] - 1) // n_chunks
```